frontend/src/hooks/useActions.ts

```diff
@@ -674,4 +685,80 @@ export const selectionToCopyTemplate = (stateIds: number[], commentIds: number[]
   }
 }
 
+// Functions for importing and exporting modules
+export const useParseModuleFile = <T>(onData: (val: T) => void, errorMessage: string, input: File, onFinishLoading: () => void, onFailedLoading: () => void) => {
+  // Read file data
+  const reader = new FileReader()
+  reader.onloadend = () => {
+    try {
+      const parse = JSON.parse
+      const fileData = parse(reader.result as string)
+      const module = {
+        ...createNewModule(),
+        ...fileData
+      }
+      onData({
+        ...module,
+        meta: {
+          ...module.meta,
+          name: module.meta.name || input?.name.split('.').slice(0, -1).join('.')
+        }
+      })
+      onFinishLoading()
+    } catch (error) {
+      showWarning(`${errorMessage}\n${error}`)
+      console.error(error)
+      onFailedLoading()
+    }
+  }
+  reader.readAsText(input)
+}
+
+export const promptLoadModuleFile = <T>(onData: (val: T) => void, errorMessage = 'Failed to parse file', accept: string, onFinishLoading = () => null, onFailedLoading = () => null) => {
+  // Prompt user for file input
+  const input = document.createElement('input')
+  input.type = 'file'
+  input.accept = accept
+  input.onchange = () => { useParseModuleFile(onData, errorMessage, input.files[0], onFinishLoading, onFailedLoading) }
+  input.click()
+}
+
+export const exportModuleFile = () => {
+  // Pull module state
+  const project = useModuleStore.getState().module
+
+  // Create a download link and use it
+  const a = document.createElement('a')
+  const file = new Blob([JSON.stringify(project, null, 2)], { type: 'application/json' })
+  a.href = URL.createObjectURL(file)
+  // File extension explicitly added to allow for file names with dots
+  a.download = project.meta.name.replace(/[#%&{}\\<>*?/$!'":@+`|=]/g, '') + '.aom'
+  a.click()
```

Comment:
This snippet is starting to get reused a lot.
Should be refactored into a function